### ISSUE TITLE
Perf / accuracy metrics comparison with Nemo for SFT scenario

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/torchtune#2760]
Do you by chance have any perf / accuracy metrics comparison with Nemo for basic fine-tuning?

Thanks :)

**Comment by pbontrager** [comment]
We don't have specific benchmark numbers for any other library. You can see some of our numbers [here](https://github.com/pytorch/torchtune/pull/2698) and try to put together a comparison yourself though.

@ebsmothers do you have any additional thoughts?

**Comment by vadimkantorov** [comment]
Basically wondering if torchtune is a good tool for some challenges like https://github.com/huggingface/open-r1 which does SFT on math reasoning traces, and then grpo...

**Comment by ebsmothers** [comment]
@vadimkantorov sorry I missed this until now. We do provide perf and memory numbers running SFT on an instruct dataset on a single A100 in our readme [here](https://github.com/pytorch/torchtune#optimization-flags). But I think it would be good for us to add similar numbers for a full node as well. Regarding reproducible logs, is there a certain set of runs you are interested in seeing here?

**Comment by RedTachyon** [comment]
Hi, I don't think there's any comprehensive benchmarking effort, but I have a bunch of curves from a heavily modified* version of this recipe.

Note that all of this is mostly intermediate benchmarks for other research, so there are some crashes, some weird noisy runs etc. But hopefully it's better than nothing. Each model was trained on 8xA100.


*The algorithm should be substantially the same, but we have an internal version that uses vLLM for data generation, so it's probably much faster than the current public version. I think we also changed advantage normalization and the length normalization of the loss. 
At some point I can get around to cleaning up and publishing the vLLM stuff, especially if there's some demand from the community.

NB:
Llama models are whichever Llama 3 variants with that size
Qwens are Qwen 2.5



GSM8k with {llama3b, llama8b, qwen3b, qwen7b}:

<img width="1386" alt="Image" src="https://github.com/user-attachments/assets/362bf99f-5f03-4446-a20c-986b442b0905" />

<img width="1397" alt="Image" src="https://github.com/user-attachments/assets/bc41f228-3dcb-4936-9fc0-d8a1fc63b2cc" />


MATH with {llama3b, llama8b, qwen3b, qwen7b}:

<img width="1396" alt="Image" src="https://github.com/user-attachments/assets/a47a8fb6-0a89-483e-8d29-025b22955a17" />

<img width="1389" alt="Image" src="https://github.com/user-attachments/assets/e9c02f61-8fe6-4c15-9886-0c22668366bf" />



DAPO training, MATH-500 evaluation with qwen7b

<img width="1376" alt="Image" src="https://github.com/user-attachments/assets/4aad145e-8c27-4bb2-8c0b-0d004d3f7d70" />
